src/caches/low_storage_rk_caches.jl

```diff
@@ -1,3 +1,13 @@
+DiffEqBase.@def lsrkcache begin
+  tmp = zero(rate_prototype)
+  williamson_condition = alg.williamson_condition
+  if williamson_condition || f isa IncrementingODEFunction
+    k = tmp
+  else
+    k = zero(rate_prototype)
+  end
+  LowStorageRK2NCache(u,uprev,k,tmp,tab)
+end
```

Comment:
I'm wondering, what are the advantages/disadvantages over using a proper function here? Or aybe just use an `@eval` loop that defines the whole `alg_cache` function for a given combination of algorithm type and tableau type? I.e., something like
```julia
for (A, T) in zip(algtypes, tableau)
    @eval begin
        function alg_cache(alg::$A, u, rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,::Val{true})
            tab = $T(real(uBottomEltypeNoUnits), real(tTypeNoUnits))
            tmp = zero(rate_prototype)
            williamson_condition = alg.williamson_condition
            if williamson_condition || f isa IncrementingODEFunction
                k = tmp
            else
                k = zero(rate_prototype)
            end
            return LowStorageRK2NCache(u,uprev,k,tmp,tab)
        end
    end
end
```